glotaran/model/dataset.py

```diff
@@ -49,96 +58,198 @@ def get_calculated_axis(self):
-        if any(not _hashable(v) for v in axis):
-            raise ValueError("Axis must be list or ndarray of hashable values")
-        self._axis[label] = axis
+        warnings.warn(DeprecationWarning("The method `dataset.get()` is deprecated, "
+                                         "use `dataset.data` instead"))
+        return self.data
 
-    def get(self) -> np.array:
-        """nd.array of shape (M,N)
+    def set(self, data: np.ndarray):
+        """
+        `Deprecated` Sets data as np.ndarray of shape (M,N)
 
-        Returns
-        -------
-        data : np.array
+        Parameters
+        ----------
+        data : np.ndarray
 
+        """
+        warnings.warn(DeprecationWarning("The method `dataset.set(data)` is deprecated, "
+                                         "use `dataset.data = data` instead"))
+        self.data = data
+
+    @property
+    def data(self) -> np.ndarray:
+        """
+        Data of the dataset as np.ndarray of shape (M,N)
 
+        Returns
+        -------
+        data:  np.ndarray
         """
         return self._data
 
-    def set(self, data):
+    @data.setter
+    def data(self, data: np.ndarray):
         """
+        Data of the dataset as np.ndarray of shape (M,N).
+        This corresponds to a measurement with M indices (i.e. wavelength, pixel)
+        and N time steps.
 
         Parameters
         ----------
-        data : np.array
-
+        data: np.ndarray
+            Data of the dataset as np.ndarray of shape (M,N)
 
         """
         if not isinstance(data, np.ndarray):
-            raise TypeError("Data must be a nd array")
+            raise TypeError("The data needs to be a ndarray")
         if len(data.shape) is not 2:
-            raise ValueError("Dataset must be 2-dimensional")
+            raise DimensionalityError("The data needs to be 2-dimensional")
         self._data = data
 
     def copy(self) -> 'Dataset':
-        """Returns a copy of the Dataset Object
+        """
+        Returns a copy of the Dataset Object
 
         Returns
         -------
         dataset : Dataset
         """
         return copy(self)
 
-    def svd(self) -> Tuple[np.array, np.array, np.array]:
-        """Returns the singular value decomposition of the dataset
-
+    def svd(self) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+        """
+        Returns the singular value decomposition of the dataset
 
         Returns
         -------
         tuple :
             (lsv, svals, rsv)
 
-            lsv : np.array
+            lsv: np.ndarray
                 left singular values
-            svals : np.array
+
+            svals: np.ndarray
                 singular values
-            rsv : np.array
+
+            rsv: np.ndarray
                 right singular values
 
         """
         lsv, svals, rsv = np.linalg.svd(self.get().T)
         return lsv, svals, rsv.T
 
 
+class HighDimensionalDataset(Dataset):
```

Comment:
This is a helper class and does not concern models. Move to somwhere in `dataio`